pkg/ingester/ingester.go

```diff
@@ -1757,6 +1756,10 @@ func (i *Ingester) LabelValues(ctx context.Context, req *client.LabelValuesReque
 		vals[i] = strings.Clone(s)
 	}
 
+	if hints != nil && hints.Limit > 0 && len(vals) > hints.Limit {
+		vals = vals[:hints.Limit]
+	}
```

Comment:
[Nit] No big deal, but I think for optimum performance, this should be moved before line 1752.